### PR TITLE
Fix transient local retransmission after participant drop [9630]

### PR DESCRIPTION
**Pull request by MiguelCompany**

### CODE REVIEW

File: include/fastdds/rtps/builtin/discovery/endpoint/EDP.h [hunk 1]
```diff
@@ -254,7 +254,8 @@ class EDP
      */
     bool unpairWriterProxy(
             const GUID_t& participant_guid,
-            const GUID_t& writer_guid);
+            const GUID_t& writer_guid,
+            bool removed_by_lease);
```
Comment (richiware) [inline include/fastdds/rtps/builtin/discovery/endpoint/EDP.h:258]: Doxygen documentation.

File: include/fastdds/rtps/reader/RTPSReader.h [hunk 2]
```diff
@@ -79,7 +79,8 @@ class RTPSReader : public Endpoint
      * @return True if correctly removed.
      */
     RTPS_DllAPI virtual bool matched_writer_remove(
-            const GUID_t& writer_guid) = 0;
+            const GUID_t& writer_guid,
+            bool removed_by_lease = false) = 0;
```
Comment (richiware) [inline include/fastdds/rtps/reader/RTPSReader.h:83]: Doxygen documentation.

File: include/fastdds/rtps/reader/RTPSReader.h [hunk 3]
```diff
@@ -267,7 +271,8 @@ class RTPSReader : public Endpoint
      */
     void remove_persistence_guid(
             const GUID_t& guid,
-            const GUID_t& persistence_guid);
+            const GUID_t& persistence_guid,
+            bool removed_by_lease);
```
Comment (richiware) [inline include/fastdds/rtps/reader/RTPSReader.h:275]: Doxygen documentation.

**Review by richiware — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2916/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/202/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2668/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2927/)

**Review by richiware — DISMISSED** [review]

**Comment by richiware** [comment]
Please, rebase this PR

**Comment by MiguelCompany** [comment]
> Please, rebase this PR

done

**Review by richiware — APPROVED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2983/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/238/)
* Mac Build Status
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2993/)

**Pull request merged**